grpcx/middleware/breaker.go

```diff
@@ -0,0 +1,38 @@
+package middleware
+
+import (
+	"context"
+
+	"github.com/msales/pkg/v3/breaker"
+	"github.com/msales/pkg/v3/stats"
+	"google.golang.org/grpc"
+)
+
+const (
+	breakerErrorKey = "breaker.error"
+	stateTag        = "state"
+)
+
+// WithBreaker adds breaker to client request.
+func WithClientBreaker(br *breaker.Breaker) grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		st, ok := stats.FromContext(ctx)
```

Comment:
What's the point of extracting the stats here? `stats.Inc` will do exactly that for you.

Reply:
Yeah, I think it's routine that's why I did it. Fixed